### ISSUE TITLE
RPGMaker MZ (1096900) [exact same problems in RPGMaker MV (363890)]

### ISSUE THREAD

**Issue by benbrockn** [ValveSoftware/Proton#4185]
# Compatibility Report
- Name of the game with compatibility issues: RPGMaker MZ
- Steam AppID of the game: 1096900

## System Information
- GPU: GTX-1080
- Driver/LLVM version: 450.66
- Kernel version: 5.8.5-arch1-1
- Link to full system information report as [System Info Gist here](https://gist.github.com/benbrockn/f80f72514e7ff6777e02212fe9fddbb8)
- Proton version: 5.0-9

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[RPGMaker MZ Gist here](https://gist.github.com/benbrockn/a45d32eeb2de7e9b72fc95291f2af53e)

## Symptoms

* "Help -> Contents" freezes the game
* Tiny white box artifact when using the tutorial
* In the first tutorial when typing in text, you have to click the text box multiple times for it to register that you are selecting the text box (have to fight it to type)
* In the first tutorial when you open the "database", the program slows down for a few seconds (regardless of hardware) and has ghosting of the cursor/tutorial window (think of old-school solitaire ghosting on Windows). It does fix itself after those few seconds.
* When doing the first tutorial, the "stone cave" icon disappears when you go to the next step. Not sure why.
* Playtest screen pops up for a second and then disappears (supposed to load your game)

## Reproduction

**Below is my current system, but I also tested it on an Ubuntu 20.04 system with the same results**

OS: EndeavourOS Linux x86_64
Kernel: 5.8.5-arch1-1
Shell: bash 5.0.18
Resolution: 2560x1440
DE: Xfce
WM: Xfwm4
CPU: AMD Ryzen 7 2700 (16) @ 3.966GHz
GPU: NVIDIA GeForce GTX 1080
Memory: 2772MiB / 32125MiB

**Comment by ell1e** [comment]
I hope this is the right place for me to add feedback:

OS: Fedora 32 x64
Kernel: 5.8.7
Shell: 5.0.17
Resolution: 1920x1080
DE: Gnome 3
WM: Gnome 3 Shell
CPU: AMD Ryzen 5
GPU: AMD R7 240 card with the amdgpu driver (using its GCN1.0 experimental vulkan support)
Memory: 16GB

I tried Proton-5.0.9 as well, and while the playtest launches for me, it has major problems:

- t takes really long to launch (multiple seconds) and once it's visible the entire playtest window is black for another 10ish seconds, and then the actual game shows a loading spinner for yet another twenty seconds. Running this playtest with native nw.js takes ~3 seconds for the same sequence
- once the playtest runs, audio is crackling horribly
- in general, it feels like the nw.js runtime (which is used for the playtest) runs with some sort of handbrake turned on, and takes 10x+ the time for any basic task and audio perf is horribly crackling and it all just runs barely. there seems to be some specific part of it that just doesn't perform right in proton that weighs everything down badly

(playtest is what happens when you press the "Play" button inside RPG Maker, it launches an [nw.js](https://nwjs.io/) instance, basically, since the games are HTML5 apps. But obviously it'll launch the windows build of nw.js - not the native one that runs just fine with none of the above issues)

Help > Contents freezes for me as well. The tutorial for some reason I can complete with no obvious issues, but maybe I misunderstood the issues with it pointed out above.

Regarding the bad playtest behavior, I also have similar issues with RPG Maker MV released games that were released on Steam for Windows only, like "Knight Bewitched" (AppId 815150, made with the RPG Maker MV tool/AppId 363890): when run with Proton-5.0.9 it too has badly crackling audio output for me, even though at least it launches fast. (RPG Maker MV is the predecessor to MZ, which uses the same nw.js-based tech for playtest and games export.)

**Comment by benbrockn** [comment]
> I hope this is the right place for me to add feedback:
> 
> OS: Fedora 32 x64
> Kernel: 5.8.7
> Shell: 5.0.17
> Resolution: 1920x1080
> DE: Gnome 3
> WM: Gnome 3 Shell
> CPU: AMD Ryzen 5
> GPU: AMD R7 240 card with the amdgpu driver (using its GCN1.0 experimental vulkan support)
> Memory: 16GB

I'm just glad someone was able to recreate it, even in a different distro (Fedora). Hopefully others will chime in

**Comment by ell1e** [comment]
It's just such a shame that [nw.js](https://nwjs.io)/the playtest of all things runs so badly, given that's the one cross-platform project that is essentially just a chromium that one would think would work. Like, how is the open-source browser component + HTML5 web app the one thing that won't run right? Pretty sad. I work around it by launching my game manually via nw.js for linux through outside means, but it does impact the workflow quite a lot that the launch button from inside RPG Maker doesn't work.

**Comment by benbrockn** [comment]
@kisak-valve 

This issue/fix may be related, but I'm unsure how to install/run/patch with an updated nwjs:

https://github.com/ValveSoftware/Proton/issues/3694

**Comment by ell1e** [comment]
Ah, an nwjs upgrade also works around it? Fascinating, that explains why my "manual" game export runs just fine (for which I fetched the latest nwjs stable at the time). But it might be difficult to convince the RPG Maker company in any reasonable amount of time to update the nwjs they ship, and by then it'll already be out with a lot of games most likely in that version, so actually addressing this in proton seems almost unavoidable.

**Comment by benbrockn** [comment]
> But it might be difficult to convince the RPG Maker company in any reasonable amount of time to update the nwjs they ship

I've spoken with RPGMaker staff and they were very short with me when I asked for tech support for Linux (because RPGMaker MV actually supported Ubuntu 14.04 according to their Steam page). They have no desire to help with Linux or Proton.

**Comment by ell1e** [comment]
I mean if you give them the fix already (nwjs update) it might be easier, but still, surely some games have been exported already that use the older version. So there doesn't really seem to be a way around fixing this on the Proton side. The black screen issue ( https://github.com/ValveSoftware/Proton/issues/3694 ) in particular, that one is pretty bad. For me it eventually resolves itself, but only after it hangs with a black window for around 30 seconds, with even the title bar in black, not responding.

**Comment by benbrockn** [comment]
Maybe you can talk to them about it since you are more knowledgeable on the subject, and the fact that I've already bugged them about Linux, lol.

Good luck!

support@rpgmakerweb.com

**Comment by ell1e** [comment]
I'm really not sure what you mean by more knowledgeable, the idea of an nwjs update came from you :flushed:  and as I just elaborated it doesn't really help with MZ games already released before that update happens. So I think the thing that makes most sense is that Proton actually fixes whatever incompatibility is going on.

**Comment by benbrockn** [comment]
I mean, I don't even know what nwjs is, or does, or how it works. You seemed to understand what that is when you said: "Fascinating, that explains why my "manual" game export runs just fine (for which I fetched the latest nwjs stable at the time).". Which is why I said you were more knowledgeable on the subject.

I only suggested that you talk to their support team because of that first point but also because you said: "So there doesn't really seem to be a way around fixing this on the Proton side.". So I figured that fixing it via Proton was a no-go.

Did I understand you correctly? I know it won't fix games released on MZ prior to MZ being patched, but it would fit all games afterwards, plus MZ was just released, so there probably aren't many games created yet.

**Comment by ell1e** [comment]
> "So there doesn't really seem to be a way around fixing this on the Proton side.". So I figured that fixing it via Proton was a no-go.

I did really say the opposite. This needs a Proton fix. (Because while using a newer nw.js seems to help, all MZ games already out will use the nw.js version Proton currently doesn't like, and that is unlikely to ever change for most of those.)

**Comment by kisak-valve** [comment]
RPG Maker MZ doesn't work on current version of proton

Issue transferred from https://github.com/ValveSoftware/Proton/issues/4722.
@whitequill posted on 2021-04-02T06:53:31:

# Compatibility Report
- Name of the game with compatibility issues: RPGMaker MZ
- Steam AppID of the game: 1096900

## System Information
- GPU: nVidia GTX 1080
- Driver/LLVM version: OpenGL version string: 4.3 (Compatibility Profile) Mesa 20.3.5
- Kernel version: 5.11.10-200.fc33.x86_64
- https://gist.github.com/whitequill/fab4d5eff6c7d0a92f5574bd14c7ef8d
- Proton version: 4.11-13

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

I can't get a proton log 
## Symptoms 
RPG Maker MZ only works on this old version of Proton (4.11-13) and only exclusively if you use beta version 1.0.2 or 1.1.1 of RPG Maker MZ.
I would like to see it running on version 6+. I want to make sure that it is given attention so it can be used in up to date versions.

There is a small issue that when running a playtest it gives an error: "Failed to initialize graphics". Otherwise MZ is completely unusable in all other versions of Proton. 

## Reproduction
Download RPG Maker MZ and try and run it with any other version of proton. it failed to run and crashed prematurely for me.

**Comment by kisak-valve** [comment]
Hello @whitequill, looking at your system information, there's two major issues with your system in addition to the previous discussion.

First, Proton uses DXVK to translate DirectX 9 / 10 / 11 to Vulkan. Your current video driver is mesa/nouveau which does not support Vulkan, making it unsupported by Proton. You can either use the nVidia proprietary driver or add `PROTON_USE_WINED3D=1 %command%` to the game's launch options to tell Proton to use wine's DirectX to OpenGL render path, however, that launch option is expect to come with a significant performance penalty and is not considered supported.

Second, you've encountered https://github.com/ValveSoftware/steam-runtime/issues/294. Starting with Proton 5.13, Proton is run inside of the Steam Linux Runtime - Soldier container environment, and that is not currently compatible with Flatpak'd Steam.

**Comment by whitequill** [comment]
I have trouble with the proprietary nVidia drivers. They don't work very well on Fedora 33 on the GTX 1080.

I don't think its a mesa/nouveau problem because MZ does work perfectly with the settings I have found, all except for the inability to play test anything. The `PROTON_USE_WINED3D=1 %command%` does fix the error of "Failed to initialize graphics". Now its just saying my browser doesn't support WebGL. I'm not sure what wine opens it in.

Its just very niche and isn't an intuitive fix. Having to set Proton to 4.11-13 and choose beta v1.0.2 or v1.1.1.
Also using Proton 4.2-9 works.

I am able to use Deployment to Windows and Web and that does work in a pinch for testing.
Better documentation or better support would be appreciated.
Per-this-ticket I'm requesting better compatibility.

**Comment by whitequill** [comment]
I have found that opening the **Animations** will suddenly crash freeze Linux completely.
It will **always** crash but, not right away.
This is after adding: `PROTON_USE_WINED3D=1 %command% ` to the Launch Options.
The only way(s) to get out is;
1. wait for the process to die (cause it will eventually)
2. hard reboot
3. ssh into your Linux system and soft reboot.
I couldn't find what process needed to be killed to un-hang the system. I did try killing a few while sshed in.

I also can't find any good way to get debug info when it hangs.

I should add that sshing in the system is functional and problems from the terminal.

**Comment by whitequill** [comment]
I have a crash report:
https://pastebin.com/dEUc0VnQ

**Comment by usokawa** [comment]
- GPU: NVIDIA GeForce RTX 4060
- Video driver version: NVIDIA 580.65.06
- Kernel version: 6.14.0-33-generic
- Proton version: proton-10.0-2e
- RPG Maker MZ version: 1.9.1

It works without crashing.
